### DISCUSSION TITLE
NextJS SSG returns 404 when I use slug params at 12.2.5

### DISCUSSION THREAD

**Discussion by nabi-chan** [vercel/next.js#40351] [Help] [answered]
Hello. I have an error with nextJS ssg with slug params, 
It always throw an 404 error at manifest.

<img width="675" alt="스크린샷 2022-09-08 22 35 58" src="https://user-images.githubusercontent.com/42037851/189136287-8fd9fc7d-827a-4c64-9143-050b2a0e40f8.png">

```typescript
// /notes/[...slug].tsx
const Notes: NextPage<Props> = ({ note }) => {
  return (
    <div className="page-content mx-auto flex py-0"
      <article className="p-8 flex flex-col gap-2">{JSON.stringify(note)}</article>
    </div>
  );
};

export const getStaticPaths: GetStaticPaths = async () => {
  return {
    paths: await noteStaticPaths(),
    fallback: false,
  };
};

export const getStaticProps: GetStaticProps = async ({ params }) => {
  const slug = (params!.slug as string[]) || [];
  const paths = slug.slice(0, -1).join("/");
  const filename = deleteMdFileExtension(slug.slice(-1)[0]);
  console.log(slug, paths, filename)

  return {
    props: {
      notes: await noteFiles("notes"),
      note: await note(paths, filename),
    },
  };
};

export default Notes;
```

*you can check full code at https://github.com/KimPinot/new-blog/blob/notes/pages/notes/%5B...slug%5D.tsx*

It seems call scripts like `./_next/static/...`

<img width="520" alt="스크린샷 2022-09-08 22 36 56" src="https://user-images.githubusercontent.com/42037851/189136541-b8435802-6d63-4c7c-82a2-8a562d7005ea.png">

How can I fix this problem?

**Comment by icyJoseph** [comment]
Kind of weird that it requires those files on the `/notes` path. IIRC normally these type of supporting requests are made from the root.

I'll checkout the code and see what's up.

### Edit

The assetPrefix had been set to `./`, which scopes requests to the current page.

**Comment by icyJoseph** [comment, marked as answer]
Aha I knew it... it is the assetPrefix that's set to `./` and since your other pages are `/something/index.tsx` you didn't experience the issue until now.

By setting it to `./` the paths become relative to the current page. Change it to `/`, in `next.config.js`, so that it is absolute.

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
  compiler: {
    removeConsole: {
      exclude: getExcludedConsole(),
    },
  },
  images: {
    loader: "custom",
  },
  env: {
    production,
  },
  assetPrefix: "/",
  experimental: {
    swcPlugins: [["next-superjson-plugin", { excluded: [] }]],
  },
};

module.exports = nextConfig;
```

**Reply by nabi-chan** [reply]
Thanks 🙇‍♀️🙇‍♀️🙇‍♀️
